docs/content/guides/developer/sui-101/using-events.mdx

```diff
@@ -36,159 +30,220 @@ To create an event in your Move modules, add the `sui::event` dependency.
-	"parsedJson": {
-		"base_asset_quantity_placed": "5000000",
-		"client_order_id": "1696545636947311087",
-		"expire_timestamp": "1696549236947",
-		"is_bid": true,
-		"order_id": "562414",
-		"original_quantity": "5000000",
-		"owner": "0xf995d6df20e18421928ff0648bd583ccdf384ab05791d8be21d32977a37dacfc",
-		"pool_id": "0xf0f663cf87f1eb124da2fc9be813e0ce262146f3df60bc2052d738eb41a25899",
-		"price": "274518000000"
-	},
-	"bcs": "4SgemkCzrqEsTHLFgMcbUtttZCf2CrEH2njjFL1rizCHzvAoYsToGrbFLffQPtGxsSt96Xr4j2SLNeLcBGKeYXDrVYWqivhf3551Mqj71DZBxq5D1Qwfgh1TKeF43Jz4b4XH1nEpkya2Pr8515vzJbHUkpP"
-}
-```
+<details>
+<summary>
+Filter events by sender
+</summary>
+{@inject: crates/sui-graphql-rpc/examples/event_connection/filter_by_sender.graphql}
+</details>
 
-## Examples
+<details>
+<summary>
+Filter events by emitting package and type
+</summary>
+{@inject: crates/sui-graphql-rpc/examples/event_connection/filter_by_emitting_package_module_and_event_type.graphql}
+</details>
 
-### Subscribe to event
+The [TypeScript SDK](https://sdk.mystenlabs.com/typedoc/modules/_mysten_sui.graphql.html) and [Rust SDK](https://mystenlabs.github.io/sui/sui_graphql_rpc_client/index.html) provide functionality to interact with the Sui GraphQL service. 
 
-This example leverages the Sui TypeScript SDK to subscribe to events the package with ID `<PACKAGE_ID>` emits. Each time the event fires, the code displays the response to the console.
+### Filtering event queries
 
-#### TypeScript
+To filter the events returned from your queries, use the following data structures.
 
-To create the event subscription, you can use a basic Node.js app. You need the Sui TypeScript SDK, so install the module using `npm install @mysten/sui` at the root of your project. In your TypeScript code, import `JsonRpcProvider` and a connection from the library.
+:::info
 
-```ts
-import { getFullnodeUrl, SuiClient } from '@mysten/sui/client';
+This set of filters applies only to event querying APIs. It differs from the filters offered for the subscriptions API (see following section). In particular, it does not support combinations like `"All": [...]`, `"Any": [...]`, `"And": [_, _]`, `"Or": [_, _]`, and `"Not": _`. 
 
-// Package is on Testnet.
-const client = new SuiClient({
-	url: getFullnodeUrl('testnet'),
-});
-const Package = '<PACKAGE_ID>';
+:::
 
-const MoveEventType = '<PACKAGE_ID>::<MODULE_NAME>::<METHOD_NAME>';
+| Query             | Description                                              | JSON-RPC Parameter Example                                                                            |
+| ----------------- | -------------------------------------------------------- | ----------------------------------------------------------------------------------------------------- |
+| `All`             | All events                                               | `{"All"}`                                                                                             |
```

Comment:
```suggestion
| `All`             | All events                                               | `{"All": []}`                                                                                             |
```

Reply:
This explains the trouble I was having with this one.